File: 10-19/Problem_10/problem10.js
let numbers = new Array(2000000)
let result = 0
numbers.fill(1)

for (let n=2; n < 2000000; n++) {
    if (!numbers[n]) continue

    result += n
    multiple = n*2
    while (multiple < 2000000) {
        numbers[multiple] = undefined
        multiple += n
    }
}


console.log(result)

File: 10-19/Problem_10/problem10.rs


fn main() {
    let mut numbers = [1; 2_000_000];
    let mut result = 0;
    let mut multiple;

    for n in 2..2_000_000 {
        if numbers[n] == 0 {
            continue
        }
        result += n;
        multiple = n*2;
        while multiple < 2_000_000 {
            numbers[multiple] = 0;
            multiple += n;
        }
    }
    
    println!("{}", result);
}